compiler/cgen.nim

```diff
@@ -1170,6 +1170,12 @@ proc genProcAux*(m: BModule, prc: PSym) =
       returnStmt = ropecg(p.module, "\treturn $1;$n", [rdLoc(res.loc)])
     elif sfConstructor in prc.flags:
       fillLoc(resNode.sym.loc, locParam, resNode, "this", OnHeap)
+      let ty = resNode.sym.typ[0]
+      for n in ty.n: #Inits default Nim fields
+        let val = n.sym.ast
```

Comment:
I don't know about how the contructor works internall. @jmgomez Would you mind giving me a cursory introduction?

This doesn't seem to be correct. You need to handle `nkRecList`, `nkRecCase`, `nkSym` separately.

Reply:
Ouch, you are right. But those should be handled with the function that I extracted, right? I mean, I would just need to extract the AST for each kind?

`constructor` besides changing the declaration (like `member` and `virtual` which doesnt affect this use case). What it does is to dont inject `result` but `this` in both Nim and C++.
Now I want to generate the inits like so:
`this->x = (NI) 10` (will change in the next push to `(*this).x = (NI)10`, but you get the idea)

Reply:
hmm at the point it reaches `genObjConstr` is already flatten like `(x:10..n:whatever)` Im just reusing `genObjConstr` and it assumes it will have that form already